GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/GameManager.cs

```diff
@@ -50,4 +59,63 @@ public bool IsInPlayPage()
     {
         return playPageCanvas.activeInHierarchy;
     }
+
+    private void PrintCarJson()
+    {
+        sedan = new Car("sedan", 500, 0, false, true);
+       
+        print(JsonUtility.ToJson(sedan));
+    }
+
+    // save game data
+    public void SaveData()
+    {
+        // update the ownership of the car
+        foreach (var car in _gameData.cars)
+        {
+            // if playerpref has car name key, the user owned the car
+            car.owned = PlayerPrefs.HasKey(car.carName);
+            Debug.Log(car.carName);
+        } 
+        System.IO.File.WriteAllText(_dataPath, JsonUtility.ToJson(_gameData, true));
+    }
+    
+    // load game data
+    private void LoadData()
+    {
+        try
+        {
+            // check if the data file exits
+            if (System.IO.File.Exists(_dataPath))
+            {
+                string contents = System.IO.File.ReadAllText(_dataPath);
+                _gameData = JsonUtility.FromJson<GameData>(contents);
+            }
+            else // if data file doesn't exist, create a default one
+            {
+                Debug.Log("Unable to read the save data, file does not exist");
+                _gameData = new GameData();
+                _gameData.cars.Add(new Car("carSedan", 500, 0, false, true));
+                _gameData.cars.Add(new Car("carTruck", 400, 0, false, false));
+                _gameData.cars.Add(new Car("carJeep", 600, 0, true, false));
+                _gameData.cars.Add(new Car("carKart", 1000, 0, true, false));
```

Comment:
I would keep these in memory (hard coded in some classes) and save the ownership (entitlement of gas, car, vip stuff) on the disk.

Reply:
Got it!